docs-chef-io/content/server/auth.md

```diff
@@ -386,6 +386,77 @@ common causes:
     minutes. This can be fixed by syncing the clock with an Network Time
     Protocol (NTP) server.
 
+## Update a User's Key Pair for Authenticating With Chef Infra Server
+
+You can update a user's key pair on Chef Infra Server with knife using either the [`knife user reregister`]({{< relref "/workstation/knife_user#reregister" >}}) subcommand, or the [`knife user key`]({{< relref "/workstation/knife_user" >}}) subcommands.
+
+### knife user reregister
+
+Use [`knife user reregister`]({{< relref "/workstation/knife_user#reregister" >}}) to regenerate an RSA key pair for a user. Knife will store the public key on the Chef Infra Server and the private key will be displayed in the standard output, or use the `--file` option to write to a named file.
+
+```sh
+knife user reregister USERNAME (options)
+```
+
+### knife user key
+
+You can list, add, edit, and delete public keys using the following subcommands:
+
+- [`knife user key create`]({{< relref "/workstation/knife_user#key-create" >}})
+- [`knife user key delete`]({{< relref "/workstation/knife_user#key-delete" >}})
+- [`knife user key list`]({{< relref "/workstation/knife_user#key-list" >}})
+- [`knife user key edit`]({{< relref "/workstation/knife_user#key-edit" >}})
+
+{{< note >}}
+
+You can't modify a public key while using that same key to authenticate with Chef Infra Server. To update a user's key pair using the `knife user key` subcommands, create a new key pair and then delete the old key pair.
+
+{{< /note >}}
+
+To update a user's key pair:
+
+1. Check the current keys associated with the user:
+
+   ```sh
+   knife user key list USERNAME
+   ```
+
+1. Create a new key pair:
+
+   ```sh
+   knife user key create USERNAME --key-name KEYNAME --expiration-date YYYY-MM-DDTHH:MM:SSZ
+   ```
+
+   Knife will open the `config.rb` or `credentials` file in your text editor. Modify the user credentials to match the newly created key pair name and expiration date.
```

Comment:
knife will not open config.rb or credentials file. it opens editor with request data which will be send in this operation. user can edit the request data before sending.

User can replace old key file with new key file, update config.rb or credentials file for the new private key to take effect.

Note: User can have multiple active key pairs and can choose one of the key {by configuring in config.rb or credentials}  to communicate with server.